### PR TITLE
344 add h5p

### LINKED ISSUE #344: Add H5P support
Add H5P user interfaces:
- Editor
- Player (for preview?)
- Custom UI to select, install, browse content types and library (who?)

### Related to

- https://github.com/graasp/graasp-plugin-h5p/issues/2
- https://github.com/graasp/graasp-plugin-h5p/issues/3

### Things to consider

- Core files cannot be packaged. How to serve original H5P core front-end files (https://github.com/h5p/h5p-php-library and https://github.com/h5p/h5p-editor-php-library) (using S3? static proxy?) and how to supply them in the build chain (no npm support)?
- Front-end security? Is iframe isolation enough?

### PR DESCRIPTION
**Pull request by dialexo**
Close #344 

Adds support for H5P import and display in compose

TODO:
- merge dependencies first, then update package.json with non-branch URLs
- implement download

### CODE REVIEW

File: src/utils/uppy.js [hunk 1]
```diff
@@ -162,3 +162,23 @@ export const configureZipImportUppy = ({
...
+}) =>
+  configureUppy({
+    itemId,
+    onComplete,
+    onFilesAdded,
+    onError,
+    onUpload,
+    restrictions: {
+      maxNumberOfFiles: 1,
+      allowedFileTypes: ['.h5p'],
```
Comment (spaenleh) [inline src/utils/uppy.js:181]: Factor this out to a constant ?

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, just a comment about some constants that can be factored out maybe ⛹🏻‍♂️

File: README.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ REACT_APP_API_HOST=http://localhost:3000
 PORT=3111
 REACT_APP_SHOW_NOTIFICATIONS=true
 REACT_APP_AUTHENTICATION_HOST=http://localhost:3001
+REACT_APP_H5P_ASSETS_HOST=https://graasp-h5p.s3.eu-central-1.amazonaws.com
+REACT_APP_H5P_CONTENT_HOST=http://localhost:3000/p
```
Comment (pyphilia) [inline README.md:15]: Same as in perform, why p, and remove link from readme.

File: src/config/constants.js [hunk 3]
```diff
@@ -48,6 +50,16 @@ export const GRAASP_EXPLORE_HOST =
...
 
+export const H5P_ASSETS_HOST =
+  ENV_H5P_ASSETS_HOST ||
+  process.env.REACT_APP_H5P_ASSETS_HOST ||
+  'http://localhost:3000/p/h5p-content';
+
+export const H5P_CONTENT_HOST =
+  ENV_H5P_CONTENT_HOST ||
+  process.env.REACT_APP_H5P_CONTENT_HOST ||
+  'http://localhost:3000/p';
```
Comment (pyphilia) [inline src/config/constants.js:61]: Maybe reuse apihost for localhost:3000

File: src/components/main/ImportH5P.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          uppy={uppy}
+          height={200}
+          width="100%"
+          proudlyDisplayPoweredByUppy={false}
+          note={t(
+            `You can upload up to FILE_UPLOAD_MAX_FILES files at a time`,
+            {
+              maxFiles: FILE_UPLOAD_MAX_FILES,
+            },
+          )}
```
Comment (pyphilia) [inline src/components/main/ImportH5P.js:89]: Since it's fixed to one, maybe we can change this note.

**Review by pyphilia — CHANGES_REQUESTED** [review]
💃

**Pull request merged**